api/viewsets.py

```diff
@@ -97,6 +97,24 @@ class InfractionViewSet(CreateModelMixin, RetrieveModelMixin, ListModelMixin, Ge
     filter_backends = (DjangoFilterBackend, SearchFilter)
     filter_fields = ('user__id', 'actor__id', 'active', 'hidden', 'type')
     search_fields = ('$reason',)
+    frozen_fields = ('id', 'inserted_at', 'type', 'user', 'actor', 'hidden')
+
+    def partial_update(self, request, *args, **kwargs):
+        instance = self.get_object()
+
+        # Ignore updates for frozen fields.
+        data = {k: v for k, v in request.data.items() if k not in self.frozen_fields}
```

Comment:
I think it would make sense to allow editing `hidden`, and throw an exception if a request tries to update one of the fields here, or does that mess with the existing update scheme?

Reply:
Editing `hidden` doesn't make sense to me and I think the bot doesn't have an edit command for it either. This is because `hidden` is only meant to prevent DMing the user initially. Editing it afterwards will not DM the user (not currently, at least).

Reply:
An exception could be raised instead of ignoring them. Presumably a 400? Look like I'd need to subclass `APIException` for that, right?

Reply:
Doesn't this influence the infractions count the user can see about himself?

Reply:
I did not know it was even possible for a user to do that.

Reply:
Oh, I thought it was. But you're right in that there's currently no way to do this via the bot.